libraries/stdlib/jvm/src/kotlin/io/files/FileTreeWalk.kt

```diff
@@ -108,11 +129,31 @@ public class FileTreeWalk private constructor(
                 } else {
                     // Proceed to a sub-directory
                     state.push(directoryState(file))
-                    return gotoNext()
+                    return gotoNextDfs()
                 }
             }
         }
 
+        private fun gotoNextBfs(): File? {
+            val walkState = state.poll() ?: return null
+            val file = walkState.root
+            if (currentDepth >= maxDepth) {
+                return file
+            }
+            // TODO("Path.getNameCount()")
+            val listFiles = file.listFiles()
+            if (listFiles != null) {
+                for (fileItem in listFiles) {
+                    if (fileItem.isDirectory) {
+                        state.add(directoryState(fileItem))
+                    } else {
+                        state.add(SingleFileState(fileItem))
+                    }
+                }
+            }
+            return file
```

Comment:
How about increment `currentDepth` already as part of this function? Then we wouldn't need to do it in `computeNext()` above.

Reply:
Agree.

Reply:
> How about increment `currentDepth` already as part of this function? Then we wouldn't need to do it in `computeNext()` above.

Done.